lib/compress.js

```diff
@@ -803,6 +805,22 @@ merge(Compressor.prototype, {
                 }
             }
 
+            function extract_args() {
+                var iife, fn = compressor.self();
+                if (fn instanceof AST_Function
```

Comment:
Turns out `collapse_vars` will not interfere with `/*@__PURE__*/` - add a test in https://github.com/mishoo/UglifyJS2/pull/2188/commits/c6fac99b8b25c4dd032f609a84b4ed5dc993a392 to improve coverage.